lib/std/fmt/scan.zig

```diff
@@ -0,0 +1,239 @@
+                    if (i < input.len and fmt_c == input[i]) {
+                        i += 1;
+                        state = .start;
+                    } else i = std.math.maxInt(usize);
+                    continue;
+                }
+
+                if (i < input.len) {
+                    if (input[i] == '-') {
+                        neg = true;
+                        i += 1;
+                    } else if (input[i] == '+') {
+                        i += 1;
+                    }
+                }
+                if (arg_i >= args.len) return error.TooFewArguments;
+                var arg = args[arg_i];
+                switch (fmt_c) {
+                    '}', 'd' => {
+                        if (i < input.len) i += try parseInt(input[i..], 10, std.ascii.isDigit, arg, neg);
+                        arg_i += 1;
+                        if (fmt_c == '}') {
+                            state = .start; // skip specifier_end state
+                            continue;
+                        }
+                    },
+                    'c' => {
+                        arg.* = input[i];
+                        i += 1;
+                        arg_i += 1;
+                    },
+                    'e' => {
+                        var digit_count: usize = 0;
+                        while (i + digit_count < input.len) : (digit_count += 1) {
+                            const c = input[i + digit_count];
+                            if (!(std.ascii.isDigit(c) or c == '.' or c == 'e' or c == 'E' or c == '-')) break;
+                        }
+                        const start_i = if (neg) i - 1 else i;
+                        arg.* = std.fmt.parseFloat(@TypeOf(arg.*), input[start_i .. i + digit_count]) catch |e| return error.ConversionFailure;
+                        i += digit_count;
+                        arg_i += 1;
+                    },
+                    'x', 'X' => {
+                        if (std.mem.startsWith(u8, input[i..], "0x") or
+                            std.mem.startsWith(u8, input[i..], "0X")) i += 2;
+                        i += try parseInt(input[i..], 16, std.ascii.isXDigit, arg, neg);
+                        arg_i += 1;
+                    },
+                    'o' => {
+                        if (std.mem.startsWith(u8, input[i..], "0o")) i += 2;
+                        i += try parseInt(input[i..], 8, isOctalDigit, arg, neg);
+                        arg_i += 1;
+                    },
+                    'b' => {
+                        if (std.mem.startsWith(u8, input[i..], "0b")) i += 2;
+                        i += try parseInt(input[i..], 2, isBinaryDigit, arg, neg);
+                        arg_i += 1;
+                    },
+                    's' => {
+                        var si: usize = 0;
+                        while (i < input.len) {
+                            if (std.ascii.isSpace(input[i]) or (fmt_i + 2 < fmt.len and input[i] == fmt[fmt_i + 2])) break;
```

Comment:
What's `input[i] == fmt[fmt_i + 2]` checking?

Reply:
i removed this behavior in new commit. it was allowing for a fmt like "{s}[{s}]" (with no spaces) to work.  without this, everything (including square brackets) until a space will be gobbled up.